### ISSUE TITLE
Sync failed.  Either your credentials are incorrect, or your account doesn't exist on the Taskserver

### ISSUE THREAD

**Issue by ArtUshak** [GothenburgBitFactory/taskwarrior#2737]
I have a taskd installed on my server with organization **Artushak** and user **KhandamirovArtiom**.

I run:

```
task sync init
```

I expect a correct synchronization of tasks to the server.

Instead, I get the following error:

```
Sync failed.  Either your credentials are incorrect, or your account doesn't exist on the Taskserver.
```

`task diagnostics` output:

```
task 2.5.3
   Platform: Linux

Compiler
    Version: 10.3.0
       Caps: +stdc +stdc_hosted +LP64 +c8 +i32 +l64 +vp64 +time_t64
 Compliance: C++17

Build Features
      CMake: 3.18.4
    libuuid: libuuid + uuid_unparse_lower
  libgnutls: 3.7.1
 Build type: None

Configuration
       File: /home/artushak/.taskrc (found), 1661 bytes, mode 100640
       Data: /home/artushak/.task (found), dir, mode 40755
    Locking: Enabled
         GC: Enabled
     Server: artushak.ru:53589
         CA: -
Certificate: /home/artushak/.task/client.cert.pem, readable, 3700 bytes
        Key: /home/artushak/.task/client.key.pem, readable, 24745 bytes
      Trust: strict
    Ciphers: NORMAL
      Creds: Artushak/KhandamirovArtiom/************************************

Hooks
     System: Enabled
   Location: /home/artushak/.task/hooks
             (-none-)

Tests
   Terminal: 323x62
       Dups: Scanned 0 tasks for duplicate UUIDs:
             No duplicates found
 Broken ref: Scanned 0 tasks for broken references:
             No broken references found
```

`taskd diagnostics` output on server:

```
taskd 1.1.0
    Platform: Linux
    Hostname: artushak.ru

Compiler
     Version: 9.3.0
        Caps: +stdc +stdc_hosted +200809 +200809 +LP64 +c8 +i32 +l64 +vp64 +time_t64
  Compliance: C++11

Build Features
       Built: Feb 13 2022 09:31:40
       CMake: 3.16.3
     libuuid: libuuid + uuid_unparse_lower
   libgnutls: 3.6.13
  Build type: release

Configuration
   TASKDDATA: /var/taskd
        root: /var/taskd (readable)
      config: /var/taskd/config (readable)
          CA: /home/artushak/ca.cert.pem
 Certificate: /etc/letsencrypt/live/artushak.ru/fullchain.pem (readable)
         Key: /etc/letsencrypt/live/artushak.ru/privkey.pem (readable)
         CRL: 
         Log: - (missing)
    PID File: /tmp/taskd.pid (missing)
      Server: artushak.ru:53589
 Max Request: 1048576 bytes
     Ciphers: 
       Trust: strict
```

**Comment by bradyt** [comment]
@ArtUshak if you are still seeing `Sync failed.  Either your credentials are incorrect, or your account doesn't exist on the Taskserver`,  can you check the following?

Consider `taskd.credentials=<org>/<user>/<key>`.

On server, check `$TASKDDATA/orgs/<org>/users/<key>/config`. Does that file exist and contain `user=<user>`?

**Comment by ArtUshak** [comment]
I have the following line in my client config (`~/.taskrc`):

```
taskd.credentials=Artushak/KhandamirovArtiom/FOOBAR
```

where `FOOBAR` is a key.

And on the server, file `$TASKDDATA/orgs/Artushak/users/FOOBAR/config` contains the following text:

```
user=KhandamirovArtiom
```

**Comment by bradyt** [comment]
I'm not sure I have any more ideas. Does the taskd process have the appropriate permissions to access that file?

When I try with deliberately incorrect credentials and `taskd config debut.tls 1` and `taskd config debug on`, all I see in the logs are as follows:

```
s: INFO Receiving 'XXXXclient: <client>
key: <credentials key>
org: <org>
protocol: v1
type: sync
user: <user>

<sync key>


' (<n> bytes)
2022-02-27 08:30:00 INFO Auth failure: org '<org>' user '<user>' unknown
s: INFO Sending 'XXXXclient: taskd <version>
code: 430
status: Access denied


' (<m> bytes)
```

That doesn't really add much information.

I think you'll need someone better at assisting with this, or some way to provide steps so that anyone can recreate the issue.

Can you check what the `INFO Auth failure` line says? Does it say "unknown"? Or "bad key"? Or something else?

These error messages seem to originate from https://github.com/GothenburgBitFactory/taskserver/blob/v1.1.0/src/Database.cpp, so that file may prove useful for ideas on interpreting the particular message you see in logs.

**Comment by ArtUshak** [comment]
Sorry, it was a permissions issue, because I created an organization as root, but run taskserver under systemd as `taskd` user.

**Comment by jimsander** [comment]
I get the same error.  My thing.  Replaced the certs using easyrsa... can connect to the server just fine, it's not a TLS problem.  You'd think permissions, but I don't see it.

Another anomoly... after changing the certs (ca, server, client and user) the pre-existing **org**  my user belonged to was no longer found, I removed it and created another by same name and updated the user key on the client side.

Are there other hidden configs/cache and debug options to see what exactly is failing on the taskd side??

```
taskd server --debug=1 --debug.tls=2
tail -f $TASKDDATA/taskd.log
```

when task sync is run, see the taskd.log is receiving the json

```
' (50147 bytes)
s: INFO Sending 'XXXXclient: taskd 1.1.0
code: 430
status: Access denied


' (58 bytes)
```
